### DISCUSSION TITLE
How to have a DB column dynamically update

### DISCUSSION THREAD

**Discussion by Metaphysics0** [supabase/supabase#11247] [Questions] [answered]
I have a simple **profiles** table, with a few columns.

I'm trying to create a simple boolean column, `is_eligible_for_listing`, which will only get set to true when other fields have values present, (`full_name`, `contact_method`)

In a rails app, I could do something like this:
```ruby
class Profile
  field :full_name, type: String, default: ""
  field :contact_method, type: ContactMethod, default: nil
  
  field :is_eligible_for_listing, type: Boolean, default: false
  
  before_save :set_listing_eligibility!

  private

  def set_listing_eligibility!
    return unless full_name_changed? || contact_method_changed?
    self.is_eligible_for_listing = full_name.present? && contact_method.present?
  end
end
```

Is there a way I can achieve this with supabase, via a trigger or function?
Ideally I'd want the trigger to only run when the two columns I'm validating change.

Thanks so much,
Ryan

**Comment by GaryAustin1** [comment, marked as answer]
You can use a generated column:
https://www.postgresql.org/docs/current/ddl-generated-columns.html
or you can do a before update and insert trigger that does something like `if (new.col1 is not null and new.col1 is not null) new.col3 = true;`
